libyul/AsmAnalysis.cpp

```diff
@@ -101,24 +100,37 @@ AsmAnalysisInfo AsmAnalyzer::analyzeStrictAssertCorrect(Dialect const& _dialect,
 std::vector<YulString> AsmAnalyzer::operator()(Literal const& _literal)
 {
 	expectValidType(_literal.type, nativeLocationOf(_literal));
-	if (_literal.kind == LiteralKind::String && _literal.value.str().size() > 32)
+	bool erroneousLiteral = false;
+	if (_literal.kind == LiteralKind::String && !_literal.value.unlimited() && _literal.value.hint() && _literal.value.hint()->size() > 32)
+	{
+		erroneousLiteral = true;
 		m_errorReporter.typeError(
 			3069_error,
 			nativeLocationOf(_literal),
-			"String literal too long (" + std::to_string(_literal.value.str().size()) + " > 32)"
-		);
-	else if (_literal.kind == LiteralKind::Number && bigint(_literal.value.str()) > u256(-1))
+			"String literal too long (" + std::to_string(formatLiteral(_literal, false).size()) + " > 32)");
+	}
+	else if (_literal.kind == LiteralKind::Number && _literal.value.hint() && bigint(*_literal.value.hint()) > u256(-1))
+	{
+		erroneousLiteral = true;
 		m_errorReporter.typeError(6708_error, nativeLocationOf(_literal), "Number literal too large (> 256 bits)");
+	}
 	else if (_literal.kind == LiteralKind::Boolean)
-		yulAssert(_literal.value == "true"_yulstring || _literal.value == "false"_yulstring, "");
+	{
+		yulAssert(_literal.value.value() == true || _literal.value.value() == false);
+		if (_literal.value.hint())
+			yulAssert(*_literal.value.hint() == "true" || *_literal.value.hint() == "false");
+	}
```

Comment:
Also, about this. I remember that I said on the call that these should be normal validations but later I realized that this just won't ever happen due to how parser works. So these are really just sanity checks.

But in that case we can drop them since `validLiteral()` covers this as well.